src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala

```diff
@@ -88,31 +82,24 @@ object SelectiveMerge {
     mergedSchema
   }
 
-
   def overrideConflictingValues(df1: DataFrame, df2: DataFrame, mergedDf: DataFrame, joinKeys: Seq[String]): DataFrame = {
     val mergedSchema = getMergedSchema(df1, df2, joinKeys)
 
     val mergedDfBuilder = mergedDf.select(
       mergedSchema.map{
         case (currColumn: Column) => {
           val colName = currColumn.expr.asInstanceOf[NamedExpression].name
-          // Column is a part of df1 and doesn't belong to the join keys.
-          if (isMemberOfDf1(df1.schema, colName) && !joinKeys.contains(colName)) {
-            val origColName = colName.substring(colRenamePrefixLen)
-            when(mergedDf(colName).isNotNull, mergedDf(colName).cast(df1.schema(origColName).dataType))
-              .otherwise(df1(origColName))
-              .alias(origColName)
+          // Belongs to DF2, override.
+          if (colName.startsWith(colRenamePrefix)) {
+            mergedDf(colName).alias(colName.substring(colRenamePrefixLen))
```

Comment:
btw, you could use split function to cut off everything before the first _ including, and then you don't need to keep the prefix length

Reply:
done